adoc/architecture-updates.adoc

```diff
@@ -0,0 +1,146 @@
+interactive. These patches are safe to be applied since they should not cause any side effect.
+
+However, some patches need the nodes to be rebooted in order to be activated. This is the
+case for example of kernel updates or some packages updates (like glibc).
+
+The nodes of the cluster have some metadata that is kept up-to-date by {productname}.
+This metadata can be used by the system administrator to answer these questions:
+
+* Does the node need to be rebooted to make some updates active?
+* Does the node have pending non interactive updates?
+* When was the check done last time?
+
+System administrators can get a quick overview of each cluster node by using one of the
+following methods.
+
+UI mode:
+
+* Open a standard kubernetes UI (eg: ​kubernetes dashboard​).
+* Click on the node.
+* Look at the annotations associated to the node.
+
+Text mode:
+
+* Go to a machine with a working kubectl (meaning I can connect to the cluster).
+* Ensure I’ve the caasp kubectl plugin installed. This is a simple statically linked binary
+that SUSE distributes as RPM.
+* Execute the `kubectl caasp nodes status` command.
+
+The output of the command will look like that:
+
+[source,bash]
+----
+NAME                OS-IMAGE     KERNEL-VERSION                CONTAINER-RUNTIME   HAS-UPDATES   HAS-DISRUPTIVE-UPDATES
+ag-master-caasp-0   SLE 15 SP1   4.12.14-1p150.12.28-default   docker://18.6.1     yes           yes
+ag-master-caasp-1   SLE 15 SP1   4.12.14-1p150.12.28-default   docker://18.6.1     yes           yes
+ag-master-caasp-2   SLE 15 SP1   4.12.14-1p150.12.28-default   docker://18.6.1     yes           yes
+ag-worker-caasp-0   SLE 15 SP1   4.12.14-1p150.12.28-default   docker://18.6.1     yes           yes
+----
+
+==== Node reboots
+
+Some updates require the node to be rebooted to make them active. As a system administrator
+I don’t have to worry about a thing.
+
+The {productname} cluster is configured by default to take advantage of ​kured​. This service look for
+nodes that have to be rebooted and, before doing the actual reboot, takes care of draining the
+node.
+
+Kured reboots one node per time. This ensures my worker nodes won’t be saturated and my
+etcd cluster will always be in a healthy state.
+
+System administrators can integrate kured stats into my prometheus instance and create alerts,
+charts and other personal customizations.
+
+System administrators can also fine tune the kured deployment to prevent its agent from
+rebooting machines where special workloads are running. For example: I can prevent kured
+from rebooting nodes running computational workloads until their pods are done. To achieve
+that I just have to instruct all my users to add special labels to the pods they don’t want to see
+interrupted due to the node being rebooted.
+
+==== Interactive upgrades
+
+Some updates might cause damages to the running workloads. For example, docker updates
```

Comment:
Not sure if we should keep this example, as CaaSP v4 is now exclusively crio based.

Reply:
I took the text from the "CaaSP evolution vNext" document. If this has changes please adjust it.